VersionComparisonClass.py

```diff
@@ -0,0 +1,91 @@
+import semantic_version as sv
+import json
+
+class CheckVersion():
+  def __init__(self, VersionRangeList, VersionToCheck):
+    """
+    Inputs 
+    - a list of versions in str format from the file generated from generate_SNYK_vulnerabilities
+      - Example: ["[3.2-alpha,3.3-beta-2)"] 
+      - Splits up by the ','. Hence, every range exists in the following format
+        - ['[3.2-alpha', '3.3-beta-2)']
+    - Version to check in the intervals
+    """
+    self.VersionRangeInput = [interval.strip() for versionRange in VersionRangeList for interval in versionRange.split(',')]
+    self.VersionToCheck = sv.Version.coerce(VersionToCheck)
+
+  def CheckVersionInRange(self):
+    """
+    Iterates through the list in self.VersionRangeInput. 
+    Treats the beginning and end of range as seperate versions and constructs Spec object.
+    Checks the given range in the Spec object. Returns True, if found.
+    Else, goes through the next set of ranges.
+    """
+    VersionRangeList = []
+    for versionPoint in self.VersionRangeInput:
+      VersionRangeList.append(self.UnpackRange(versionPoint))
+      if len(VersionRangeList) == 2:
+        if self.VersionToCheck in sv.Spec(','.join(VersionRangeList)):
+          return True
+        VersionRangeList = []
+    return False
+
+  def UnpackRange(self, versionPoint):
+    """
+    Takes the versions with the corresponding brackets and unpacks it as a mathematical notation
+    This is done to comply with the semantic_version Spec object.
+    """
+    if len(versionPoint) == 1: # [,XXX]
+      if versionPoint[0] == '[' or versionPoint[0] == '(':
+        versionPoint = versionPoint[0] + '0'
+      else:
+        versionPoint = '10000000' + versionPoint[0] #Arbitrarily huge number to satisfy open ended upper limit.
+
+    if versionPoint[0] == '[':
+      return '>=' + str(sv.Version.coerce(versionPoint[1:]))
+    elif versionPoint[0] == '(':
+      return '>' + str(sv.Version.coerce(versionPoint[1:]))
+    elif versionPoint[-1] == ']':
+      return '<=' + str(sv.Version.coerce(versionPoint[:-1]))
+    elif versionPoint[-1] == ')':
+      return '<' + str(sv.Version.coerce(versionPoint[:-1]))
+    else:
+      raise Exception("Range not recongized")
+
+if __name__ == "__main__":
+  """
+  Test
+  """
+  VersionRange = ['[3.2-alpha,3.3-beta-2)']
```

Comment:
Does it work with versions like 'jenkins-1.2.3.4' (with a prefix)?

Reply:
Thanks for pointing that out. 
I made a fix for that see 374fa11, ce57ecb. 
This is more like a bandaid fix. Should work in theory. Also working in the test cases. Needs more testing to check the efficacy of the fix.
It converts the prefix into a suffix which then gets treated as a regular patch name, so think: Jenkins < Prototype. I've tested this with some variations like Jenkins-1.2.3-alpha which works. 
What do you think?

Of course, the good way to fix this would be to comb through the github tags and work with them. But even those would have to be worked in a similar manner, I think.